src/app/admin/modules/data-management/components/data-graph-child/data-graph-child.component.ts

```diff
@@ -0,0 +1,245 @@
+import { Component, OnInit , Input, ViewChild, ElementRef} from '@angular/core';
+import * as echarts from 'echarts';
+import { EChartOption } from 'echarts';
+import { Graph } from 'src/app/shared/interfaces/graph';
+import { GraphData } from 'src/app/shared/interfaces/graph-data';
+import { PieGraphData } from 'src/app/shared/interfaces/pie-graph-data';
+
+@Component({
+  selector: 'app-data-graph-child',
+  templateUrl: './data-graph-child.component.html',
+  styleUrls: ['./data-graph-child.component.css']
+})
+
+
+export class DataGraphChildComponent implements OnInit {
+  @ViewChild('myCharts') myCharts: ElementRef;
+
+  @Input() titleGraphNames?: Graph[];
```

Comment:
This component doesn't need to know the options list, you can send the name as input intead.